sympy/solvers/solveset.py

```diff
@@ -1325,6 +1329,85 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _is_bivariate(f, symbol):
+    r"""
+    Helper to check if the equation takes bivariate form.
+    If the equation has exactly two generators then we can
+    say it to be in bivariate form.
+    """
+    poly = f.as_poly()
+    gens = _filtered_gens(poly, symbol)
+
+    return True if len(gens) == 2 else False
+
+
+def _solve_as_bivariate(lhs, rhs, symbol, domain):
+    r"""
+    Helper solver to solve equations of bivariate form.
+    """
+    result = ConditionSet(symbol, Eq(lhs - rhs, 0), domain)
+
+    poly = lhs.as_poly()
+    gens = _filtered_gens(poly, symbol)
+    gpu = bivariate_type(lhs - rhs, *gens)
+    if gpu is not None:
+        g, p, u = gpu
+        inversion = _transolve(g - u, symbol, domain)
```

Comment:
Calling `_transolve` here again is causing recursion for some equations like: `exp(x) + sin(x)`, `(sin(x)/2 - cos(x)/2)*exp(x) + (sin(x)/2 + cos(x)/2)*exp(x) - 1`. The call expects that `transolve` will be able to solve equation in this pass, but if it is unable to solve, the unchanged equation get into infinite loop.
We probably need a  better way to handle bivariate type equation or maybe improve/add some checks within `transolve`.

I would suggest you to go through the `solve` module and see how it handles such equations.

Reply:
Ok @Yathartha22 I m on it.